src/main/java/seedu/address/logic/Logic.java

```diff
@@ -48,6 +53,13 @@ public interface Logic {
      */
     ObservableList<Reminder> getReminderList();
 
+    /**
+     * Returns a map of delivery job list in the week
+     */
+    Map<LocalDate, ArrayList<ArrayList<DeliveryJob>>> getWeekDeliveryJobList();
```

Comment:
the return type seems to be a complex structure to work with, maybe you can try encapsulate it out?